subsys/bluetooth/host/Kconfig

```diff
@@ -173,6 +173,15 @@ config BT_SETTINGS
 	  instead of each subsystem doing it independently.
 
 if BT_SETTINGS
+config BT_SETTINGS_CCC_LOAD_ON_DEMAND
+	bool "Load CCC values from settings when peer connects"
```

Comment:
I would default this y since normally devices have less ram then flash.

Reply:
I don't mind either ways, it might depend on how fast the setting backend can return a single value. If our existing backends have to load and parse the entire file to find a single setting, some might prefer to keep it disabled by default.
It would be nice to have @nvlsianpu 's opinion on how efficient existing backends are to load single settings before enabling the option by default.

Reply:
Will do and update the PR as per @rakons 's comment below.

Reply:
I clarifies my position in the main conversation. This is OK now here.